### PR TITLE
Page Close event

### LINKED ISSUE #390: page 'close' event
https://github.com/kblok/puppeteer/pull/3/commits/d46e157f19ecce1622cabfc25b2c859a6d37ce4f

### PR DESCRIPTION
**Pull request by kblok**
Closes #390

### CODE REVIEW

File: lib/PuppeteerSharp.Tests/PageTests/Events/CloseTests.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        {
+        }
+
+        [Fact]
+        public async Task ShouldWorkWithWindowClose()
+        {
+            var newPageTaskSource = new TaskCompletionSource<Page>();
+            Browser.TargetCreated += async (sender, e) =>
+            {
+                if (!newPageTaskSource.Task.IsCompleted)
```
Comment (Meir017) [inline lib/PuppeteerSharp.Tests/PageTests/Events/CloseTests.cs]: can't we just use `TrySetResult`, this way only the first call will succeed

File: lib/PuppeteerSharp/Page.cs [hunk 2]
```diff
@@ -597,10 +604,10 @@ public Task ExposeFunctionAsync<TResult>(string name, Func<TResult> puppeteerFun
         /// <param name="puppeteerFunction">Callback function which will be called in Puppeteer's context.</param>
         /// <remarks>
         /// If the <paramref name="puppeteerFunction"/> returns a <see cref="Task"/>, it will be awaited.
-        /// Functions installed via <see cref="ExposeFunctionAsync{T1, T2, T3, TResult}(string, Func{TResult})"/> survive navigations
+        /// Functions installed via <see cref="ExposeFunctionAsync{T1, T2, T3, TResult}(string, Func{T1, T2, T3, TResult})"/> survive navigations
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: nice catch

File: lib/PuppeteerSharp.Tests/PageTests/Events/CloseTests.cs (hunk 1 above)
Comment (kblok) [reply lib/PuppeteerSharp.Tests/PageTests/Events/CloseTests.cs]: Why didn't you teach me about `TrySetResult` before? :joy:

Comment (Meir017) [reply lib/PuppeteerSharp.Tests/PageTests/Events/CloseTests.cs]: I was sure we used it somewhere in the code...

**Comment by kblok** [comment]
Done @Meir017

**Pull request merged**